core/vpn/src/network.rs

```diff
@@ -577,6 +590,11 @@ impl Handler<Egress> for Vpn {
     fn handle(&mut self, msg: Egress, _: &mut Self::Context) -> Self::Result {
         let frame = msg.event.payload.into_vec();
 
+        #[cfg(feature = "packet-trace-enable")]
```

Comment:
If you must use `#[cfg(feature = "packet-trace-enable")]` anyway, so why `packet_trace_maybe` has `enable` feature inside? IMO it should be either of these, not combined, because they are redundant and causing unnecessary chaos.
My proposals:
1. `enable` in `ya_packet_trace`: make crate in such way that it will be unnecessary to use redundant `#[cfg(feature`
2. Delete `enable` in `ya_packet_trace`: and write `#[cfg(feature` when it is used in our code

Reply:
After some consideration, I disagree. I cannot modify the `ya_packet_trace` in a way which will make this uses redundant, because this is related to the lifetime of data that we're tracing in any way other than providing a `ya_packet_trace::copy` for making a conditional copy, which has way more cognitive load than `#[cfg(..)]`.

Option two is no good, because I'd be adding about 20 additional `#[cfg(..)]` just because of the 3 you're concerned about.

I assumed the naming would show that the feature-gated copies (`packet-trace-enable`) and macro invocations (`packet_trace[_maybe]!`) are tightly coupled, but to be sure I've added a clarifying comment to the copies, as they're a kind of an edge case.